File: SRKeyboardDemo/SRKeyboardDemo/ViewController.m
//
//  ViewController.m
//  SRKeyboardDemo
//
//  Created by Sariel's Mac on 15-5-24.
//  Copyright (c) 2015年 Sariel. All rights reserved.
//

#import "ViewController.h"
#import <SRKeyboard/SRKeyboard.h>

@interface ViewController ()

@property (weak, nonatomic) IBOutlet EmoticonsTextView *textView;

@property (strong,nonatomic) SRKeyboardController *emoticonKeyboardVC;

@end

@implementation ViewController

- (SRKeyboardController *)emoticonKeyboardVC {
    if (_emoticonKeyboardVC == nil) {
        __weak ViewController *weakSelf = self;
        _emoticonKeyboardVC = [[SRKeyboardController alloc]initWithEmoticonBlock:^(Emoticons *emoticons) {
            [weakSelf.textView insertEmoticon:emoticons];
        }];
    }
    return _emoticonKeyboardVC;
}

-(void)dealloc{
    NSLog(@"----88-----");
}

- (void)viewDidLoad {
    [super viewDidLoad];
    //添加子控制器
    [self addChildViewController:self.emoticonKeyboardVC];
    //设置textView的输入视图
    self.textView.inputView = self.emoticonKeyboardVC.view;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: SRKeyboardSwift/SRKeyboardSwift/ViewController.swift
//
//  ViewController.swift
//  SRKeyboardSwift
//
//  Created by xl_bin on 15/6/6.
//  Copyright (c) 2015年 Sariel. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var textView: EmoticonsTextView!
    @IBOutlet weak var textField: EmoticonsTextField!
    
    // 控制器对 keyboardVC 是强引用
    lazy var keyboadVC: EmoticonKeyboardViewController = {
        // 闭包中会对 self 强引用
        weak var weakSelf = self
        return EmoticonKeyboardViewController(selectedEmoticon: { (emoticon) -> () in
            // 调用方法，同样会产生强引用，在 OC 中同样需要注意
            weakSelf?.textView?.insertEmoticon(emoticon)
            weakSelf?.textField?.insertEmoticon(emoticon)
        })
        }()
    
    deinit {
        println("88")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        // 设置 UI
        setupUI()
    }
    
    private func setupUI() {
        // 1. 添加子控制器
        addChildViewController(keyboadVC)
        
        // 2. 设置 textView 输入视图 - 键盘的视图
        textView?.inputView = keyboadVC.view
        textField?.inputView = keyboadVC.view
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func show(sender: AnyObject) {
        
        println(self.textView.fullText())
    }

}

